dspace-server-webapp/src/main/java/org/dspace/app/rest/submit/step/ShowIdentifiersStep.java

```diff
@@ -0,0 +1,170 @@
+ *
+ * @author Kim Shepherd
+ */
+public class ShowIdentifiersStep extends AbstractProcessingStep {
+
+    private static final Logger log = LogManager.getLogger(ShowIdentifiersStep.class);
+
+    @Autowired(required = true)
+    protected HandleService handleService;
+    @Autowired(required = true)
+    protected ContentServiceFactory contentServiceFactory;
+
+    /**
+     * Override DataProcessing.getData, return data identifiers from getIdentifierData()
+     *
+     * @param submissionService The submission service
+     * @param obj               The workspace or workflow item
+     * @param config            The submission step configuration
+     * @return                  A simple DataIdentifiers bean containing doi, handle and list of other identifiers
+     */
+    @Override
+    public DataIdentifiers getData(SubmissionService submissionService, InProgressSubmission obj,
+                                   SubmissionStepConfig config) throws Exception {
+        // If configured, WorkspaceItemService item creation will also call IdentifierService.register()
+        // for the new item, and the DOI consumer (if also configured) will mint or delete DOI entries as appropriate
+        // while the item is saved in the submission / workflow process
+
+        // This step simply looks for existing identifier data and returns it in section data for rendering
+        return getIdentifierData(obj);
+    }
+
+    /**
+     * Get data about existing identifiers for this in-progress submission item - this method doesn't require
+     * submissionService or step config, so can be more easily called from doPatchProcessing as well
+     *
+     * @param obj   The workspace or workflow item
+     * @return      A simple DataIdentifiers bean containing doi, handle and list of other identifiers
+     */
+    private DataIdentifiers getIdentifierData(InProgressSubmission obj) {
+        Context context = getContext();
+        DataIdentifiers result = new DataIdentifiers();
+        // Load identifier service
+        IdentifierService identifierService =
+                IdentifierServiceFactory.getInstance().getIdentifierService();
+        // Attempt to look up handle and DOI identifiers for this item
+        String[] defaultTypes = {"handle", "doi"};
+        List<String> displayTypes = Arrays.asList(configurationService.getArrayProperty(
+                "identifiers.submission.display",
+                defaultTypes));
+        result.setDisplayTypes(displayTypes);
+        String handle = identifierService.lookup(context, obj.getItem(), Handle.class);
+        DOI doi = null;
+        String doiString = null;
+        try {
+            doi = IdentifierServiceFactory.getInstance().getDOIService().findDOIByDSpaceObject(context, obj.getItem());
+            if (doi != null && !DOIIdentifierProvider.MINTED.equals(doi.getStatus())
+                    && !DOIIdentifierProvider.MINTED.equals(doi.getStatus())) {
```

Comment:
Minor: not clear to me this third statement

Reply:
typo - should have excluded both MINTED and DELETED statuses when populating the identifier data section